src/core/Group.cpp

```diff
@@ -677,6 +678,9 @@ Entry* Group::findEntryByPathRecursive(const QString& entryPath, const QString&
             || (!entryPath.startsWith("/") && entry->title() == entryPath)) {
             return entry;
         }
+        if (entryPath == entry->uuid().toString(QUuid::Id128)) {
+            return entry;
+        }
```

Comment:
Search by path is not the same as searching by UUID and getting a UUID match is quite unexpected. This should be pulled out into a separate method.

Reply:
Scrolled up just a little bit and saw the method already exists.
Added it after the 1st attempt to find by path fails.

Reply:
Created the separate method.